testsuite/test_cim_operations2.py

```diff
@@ -0,0 +1,142 @@
+
+        # good cases
+        self._run_single(b'<V>a</V>', True)
+        self._run_single(b'<V>a\tb\nc\rd</V>', True)
+        self._run_single(b'<V>a\x09b\x0Ac\x0Dd</V>', True)
+        self._run_single(b'<V>a\xCD\x90b</V>', True)             # U+350
+        self._run_single(b'<V>a\xE2\x80\x93b</V>', True)         # U+2013
+        self._run_single(b'<V>a\xF0\x90\x84\xA2b</V>', True)     # U+10122
+
+        # invalid XML characters
+        if self.VERBOSE:
+            print("From here on, the only expected exception is ParseError "
+                  "for invalid XML characters...")
+        self._run_single(b'<V>a\bb</V>', False)
+        self._run_single(b'<V>a\x08b</V>', False)
+        self._run_single(b'<V>a\x00b</V>', False)
+        self._run_single(b'<V>a\x01b</V>', False)
+        self._run_single(b'<V>a\x1Ab</V>', False)
+        self._run_single(b'<V>a\x1Ab\x1Fc</V>', False)
+
+        # correctly encoded but ill-formed UTF-8
+        if self.VERBOSE:
+            print("From here on, the only expected exception is ParseError "
+                  "for ill-formed UTF-8 Byte sequences...")
+        # combo of U+D800,U+DD22:
+        self._run_single(b'<V>a\xED\xA0\x80\xED\xB4\xA2b</V>', False)
+        # combo of U+D800,U+DD22 and combo of U+D800,U+DD23:
+        # pylint: disable=line-too-long
+        self._run_single(
+            b'<V>a\xED\xA0\x80\xED\xB4\xA2b\xED\xA0\x80\xED\xB4\xA3</V>',
+            False)
+
+        # incorrectly encoded UTF-8
+        if self.VERBOSE:
+            print("From here on, the only expected exception is ParseError "
+                  "for invalid UTF-8 Byte sequences...")
+        # incorrect 1-byte sequence:
+        self._run_single(b'<V>a\x80b</V>', False)
+        # 2-byte sequence with missing second byte:
+        self._run_single(b'<V>a\xC0', False)
+        # 2-byte sequence with incorrect 2nd byte
+        self._run_single(b'<V>a\xC0b</V>', False)
+        # 4-byte sequence with incorrect 3rd byte:
+        self._run_single(b'<V>a\xF1\x80abc</V>', False)
+        # 4-byte sequence with incorrect 3rd byte that is an incorr. new start:
+        self._run_single(b'<V>a\xF1\x80\xFFbc</V>', False)
+        # 4-byte sequence with incorrect 3rd byte that is an correct new start:
+        self._run_single(b'<V>a\xF1\x80\xC2\x81c</V>', False)
+
+
+class TestGetRsltParams(object):
+    """Test WBEMConnection._get_rslt_params method."""
+
+    @pytest.mark.parametrize(
+        'irval, ec, eos, eosr', [[None, u'contextblah', u'False', False],
+                                 [None, "", u'True', True],
+                                 [None, u'contextblah', u'True', True]]
+    )
+    @staticmethod
+    def test_pass(irval, ec, eos, eosr):
+        """Test good combinations of EndOfSequence and EnumerationContext"""
+        result = [
+            (u'IRETURNVALUE', {}, irval),
+            (u'EnumerationContext', None, ec),
+            (u'EndOfSequence', None, eos)
```

Comment:
Your test always has both elements. Because the tested function allows one of the elements to be missing and has defaults, that should also be tested.
I suggest to extend the structure of the parametrization data for that. Maybe the `result` list that you have in the function should be in the parametrization data, plus `ec_exp` and `eos_exp` for the expected values.